jekyll/_cci2/orbs-faq.md

```diff
@@ -0,0 +1,116 @@
+
+### Build Error When Testing Locally
+
+* Question: Why do I get the following error when testing locally:
+
+```circleci build -c .circleci/jobs.yml --job test
+```
+
+```Error:
+You attempted to run a local build with version 2.1 of configuration.
+```
+
+* Answer: To resolve this error, run `circleci config process` on your configuration and then save that configuration to disk. You then should run `circleci build` (this has now been moved to `circleci local execute`) against the processed configuration.
+
+<!---
+### Environment Variables Not Being Passed at Runtime
+
+Occasionally, when you try to convert a configuration to a 2.0 compatible format, environment variables may not be passed at runtime. For example, if you create a simple configuration in your GitHub repository (for example `https://github.com/yourusername/circle-auto/blob/master/.circleci/echo.yml`) and then call the config using:
+
+```export AUTO_FILE=/Users/yourusername/Desktop/apkpure_app_887.apk
+export AUTO_DIR=.
+circleci build -c .circleci/echo.yml --job test
+```
+
+The config shows:
+
+```#!bin/bash -eo pipefail
+echo file $(AUTO_FILE) dir $(AUTO_DIR)
+file directlySuccess!
+```
+Upon execution, you may see the following response:
+
+.circleci/echo.yml
+
+```version: 2
+jobs:
+  build:
+    docker:
+    - image: circleci/openjdk:8-jdk
+    steps:
+    - checkout
+  test:
+    docker:
+    - image: circleci/openjdk:8-jdk
+    environment:
+    - TERM: dumb
+    steps:
+    - checkout
+    - run:
+        command: "echo file ${AUTO_FILE} dir ${AUTO_DIR}"
+workflows:
+  version: 2
+  workflow:
+    jobs:
+    - build
+    - test```
+
+yourusername/circle-autoAdded by GitHub
+```
+--->
+
+### Standard Process To Access Previously Stored Artifacts
+
+* Question: When developing plugins (especially ones that test mobile binary), is there a standard way to access previously stored artifacts and store them consistently?
+
+* Answer: If you are simply passing elements across jobs, CircleCI recommends using our workspaces rather than artifacts because workspaces can be saved in one job and restored another. Artifacts are designed for elements that need to be retrieved after execution is over. There is an API for artifacts; however, CircleCI does not have the type of first-class retrieval of "the artifact created in the upstream job" developed yet.
+
+### Logging Outputs
+
+* Question: Is there a standard way to to log output? For example, Jenkins plugins provide console links to show the log output and provides hooks to log those messages. It is possible to log to stdout, but is there a better way to log those log messages.
+
+* Answer: In CircleCI, whatever steps are run are logged, so any output from those steps will appear in the console. You can shoe to `echo` things directly. CircleCI does not have a separate logging facility outside the console output.
+
+### Failing Builds
+
+Question: How can I intentionally fail a build that invokes an orb from within an orb?
```

Comment:
Better to use the word "job" instead of "build" in this question -- but more importantly, I'm not sure why this is only about orbs in orbs -- using a non-zero exit or `step halt` are general techniques to use in config and in orbs but aren't specific to "orbs in orbs"

Reply:
this just came up in the question about orbs, I have changed it to job instead of build